.github/workflows/test.yml

```diff
@@ -0,0 +1,30 @@
+name: Pre-commit
+
+on:
+  pull_request:
+    branches: [main]
+
+concurrency:
+  group: "test"
+  cancel-in-progress: false
+
+jobs:
+  pre-commit:
+    name: Pre-Commit
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+      - name: Install Poetry Package
+        run: |
+          pip install --upgrade pip
+          pip install poetry==1.3.2
+          poetry config virtualenvs.create false
+          poetry install --no-interaction --with dev
+          pip install pre-commit
```

Comment:
Can you please add pre-commit to the dev dependencies so it doesn't need to be installed separately.

Reply:
Can you also please add ruff to the dev dependencies and the following line for ruff. I think this may undo a lot of the line length related changes.

```toml
[tool.ruff]
line-length = 120
```

Reply:
Added in dev dependencies